### ISSUE TITLE
url_regex doesn't correctly match all allowed characters in a URI

### ISSUE THREAD

**Issue by jonathanunderwood** [samuelcolvin/pydantic#1142]
# Bug

```python
from pydantic.networks import url_regex
m=url_regex.match("postgresql+psycopg2://postgres:postgres@localhost:5432/hatch")
parts = m.groupdict()
parts
```
produces the following incorrect matching:

```python
{'scheme': None,
 'user': 'postgresql+psycopg2',
 'password': '//postgres:postgres',
 'ipv4': None,
 'ipv6': None,
 'domain': 'localhost',
 'port': '5432',
 'path': '/hatch',
 'query': None,
 'fragment': None}
```

This is with Python 3.7.5 on OSX, pydantic version 1.2

**Comment by jonathanunderwood** [comment]
Note that according to Appendix A of [rfc2396](http://www.ietf.org/rfc/rfc2396.txt) a URI scheme is defined as:

```
scheme        = alpha *( alpha | digit | "+" | "-" | "." )
```

**Comment by samuelcolvin** [comment]
Thanks for reporting, please could you submit a PR for this?

**Comment by jonathanunderwood** [comment]
I'll look into it. Will need to get work clearance etc tho, so it might take a little while.

**Comment by samuelcolvin** [comment]
Thank you, let me know if it's too much work to be worth it and/or urgent, if so I can do it.

**Comment by jonathanunderwood** [comment]
It would require legal review, so if you didn't mind doing it, that would be very much appreciated.

**Comment by samuelcolvin** [comment]
wow :sleeping:, there are advantages to working for a startup.

I'll do it.

**Comment by samuelcolvin** [comment]
Fixed via #1148, please review and check that will work for you.
